hashes/src/sha256.rs

```diff
@@ -155,16 +155,17 @@ impl Midstate {
 
     /// Copies a byte slice into the [`Midstate`] object.
     pub fn from_slice(sl: &[u8]) -> Result<Midstate, Error> {
-        if sl.len() != Self::LEN {
-            Err(Error::InvalidLength(Self::LEN, sl.len()))
-        } else {
+        if sl.len() == Self::LEN {
             let mut ret = [0; 32];
             ret.copy_from_slice(sl);
             Ok(Midstate(ret))
+        } else {
+            Err(Error::InvalidLength(Self::LEN, sl.len()))
         }
     }
 
     /// Unwraps the [`Midstate`] and returns the underlying byte array.
+    #[must_use = "returns inner type without modifying self"]
     pub fn to_byte_array(self) -> [u8; 32] { self.0 }
```

Comment:
Nobody would ever expect a `to_x` function to modify `self`.

Reply:
This is a perfect example of my concern that I wrote useless messages, we should just have a plain `#[must_use]` for all `to_` methods, right?

Reply:
Hmm, yeah, `#[must_use]` without messages would be fine IMHO.